lib/reverse_sentence.rb

```diff
@@ -1,6 +1,48 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
-def reverse_sentence(my_sentence)
-  raise NotImplementedError
-end
+# Time complexity: O(n^2), because there are two nested loops that are dependent
+# on size of the same input.
+# Space complexity: O(1), because no data structures are created. I only reference
+# elements in the string by index.
+def reverse_sentence(sentence)
+  return nil if sentence == nil
+
+  i = 0
+  j = sentence.length - 1
+  while i < j
+    temp = sentence[i]
+    sentence[i] = sentence[j]
+    sentence[j] = temp
+    
+    i += 1
+    j -= 1
+  end
```

Comment:
Notice that you're reversing the whole string here and then below reversing substrings.  Could you break this code into a helper method?